fw/thread.go

```diff
@@ -430,32 +449,41 @@ func (t *Thread) processIncomingData(packet *defn.Pkt) {
 	strategy := t.strategies[strategyName.Hash()]
 
 	if len(pitEntries) == 1 {
+		// When a single PIT entry matches, we pass the data to the strategy.
+		// See alternative behavior for multiple matches below.
+		pitEntry := pitEntries[0]
+
 		// Set PIT entry expiration to now
-		table.SetExpirationTimerToNow(pitEntries[0])
+		table.SetExpirationTimerToNow(pitEntry)
 
 		// Invoke strategy's AfterReceiveData
 		core.LogTrace(t, "Sending Data=", packet.Name, " to strategy=", strategyName)
-		strategy.AfterReceiveData(packet, pitEntries[0], *packet.IncomingFaceID)
+		strategy.AfterReceiveData(packet, pitEntry, *packet.IncomingFaceID)
 
 		// Mark PIT entry as satisfied
-		pitEntries[0].SetSatisfied(true)
+		pitEntry.SetSatisfied(true)
 
 		// Insert into dead nonce list
-		for _, outRecord := range pitEntries[0].OutRecords() {
+		for _, outRecord := range pitEntry.OutRecords() {
 			t.deadNonceList.Insert(data.NameV, outRecord.LatestNonce)
 		}
 
 		// Clear out records from PIT entry
-		pitEntries[0].ClearOutRecords()
+		// TODO: NFD dev guide specifies in-records should not be cleared - why?
```

Comment:
Don't know either. I think the PIT record is completely converted into a CS record in DPDK.